rules/execute-command.js

```diff
@@ -132,33 +131,39 @@ async function runCommand(command, args = []) {
   }
 }
 
-function smuggleLockfile(rootDir) {
+function smuggleYarnProjectFiles(rootDir) {
   // Support non-hermetic yarn.lock/.pnp.cjs/.pnp.loader.mjs smuggling into sandbox
   // This allows a repo to use yarn plugins to implement custom change detection
   // to avoid invalidating top-level cache in cases where yarn.lock is touched but only affects certain projects
   // See: Lockfile delegation section in README
 
-  try {
-    const realRoot = dirname(realpath(`${rootDir}/package.json`)); // FIXME: technically a target may not depend on this file, so it's not actually guaranteed to exist, even though it usually does
+  // @see {@link https://github.com/yarnpkg/berry/blob/07cf3531002a3f25fd6309e05d0cf1b233c59cd4/packages/yarnpkg-fslib/sources/path.ts#L21-L35}
+  const yarnProjectFiles = [
+    'yarn.lock',
+    '.pnp.data.json',
+    '.pnp.loader.mjs',
+    '.pnp.cjs',
+    '.yarnrc.yml',
+    '.yarnrc',
+    'package.json',
```

Comment:
That's the whole purpose of this PR is to make it possible to exclude root `package.json` file from the bazel dependency graph... This decision definitely was not an easy one to make, but I figured that this may be inline with our current "recommended" setup without inducing some major refactoring (given we're already making a compromise on hermeticity). 

My reasoning for this is that typical changes to the root `package.json` have very little impact on the build process of the underyling workspaces, as it mainly serves as root manifest file used during install and:
* specifies list of subprojects (workspaces), and therefore is updated often when a new workspace is added / removed
* tells how dependencies should be resolved / installed in context of the whole monorepo (e.g., the [`resolutions`](https://yarnpkg.com/configuration/manifest#resolutions) and [`dependenciesMeta`](https://yarnpkg.com/configuration/manifest#dependenciesMeta) fields)
* declares some one-off scripts that are useful in context of managing / supporting the monorepo (in which case it may require installing it's own dependencies)

I'm assuming that if any of these values did change, this would be reflected in the generated lockfile local to the workspace (e.g., `workspace-deps.txt`). And users still have an option to opt-out from this behavior by explicitly defining their dependency on these top-level files in their `BUILD.bazel` file.

Reply:
If we're going to smuggle package.json into sandboxes, we probably need to ensure that the workspace-deps yarn plugin reflects changes to package.json that are relevant to their respective targets. I think for resolutions and patches we're covered. I don't think we're covered for changes to `workspaces` (and I'm not sure we can ever be)

We could willfully ignore potential risk of landing broken `workspaces` fields, as the risk is generally low, but it's not a zero risk thing.

Reply:
> I don't think we're covered for changes to workspaces (and I'm not sure we can ever be)

Could you elaborate on this point a bit?

Reply:
Here're couple things I could come up with that would not be reflected in the individual workspace lockfile (depends on implementation):
* We don't differentiate between local workspace vs. npm when generating workspace-deps.txt, so we may not catch when removing/adding a workspace changed how certain packages end up being resolved _(low risk / fixable)_
* Some package.json scripts may call out to another workspace via `yarn workspace PACKAGE_X run CMD_Y`, in which case removing a workspace may break the command.. Ideally folks should have declared the dependency on that other workspace. Otherwise I don't think it's a good idea to support a use-case such as this.. _(low risk / fixable on user-end)_
* I guess changing `packageExtensions` / adding `resolutions` may also introduce some changes deep in the dep-tree, which would not be reflected in workspace-deps.txt as we compute a shallow list of all packages found under the workspace's tree root _(low risk / fixable)_